### PR TITLE
Dev-142: Add scripts to replace Old Majors with New Majors

### PR DESCRIPTION
**Pull request by RosaGao**

### CODE REVIEW

File: data/replaceOldMajors.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+async function replaceOldWithNew() {
+  connect();
+
+  const users = await User.find({});
+  for (let user of users) {
+    for (let plan_id of user.plan_ids) {
+      let plan = await Plan.findById(plan_id).exec();
+      if (plan !== null && plan.majors != null && plan.majors.length > 0) {
+        // console.log("before: " + plan.majors);
+        for (let ind in plan.majors) {
```
Comment (mpark63) [inline data/replaceOldMajors.js:14]: i like the use of index!

File: data/replaceOldMajors.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          } else if (major.includes("Minor Computer Science (OLD - Pre-2021)")) {
+            plan.majors[ind] = "Minor Computer Science";
+          } else if (major.includes("Minor Computer Science (NEW - 2021 & after)")) {
+            plan.majors[ind] = "Minor Computer Science";
+          } else if (major.includes("Minor Applied Mathematics & Statistics (OLD - Pre-2021)")
+          || major.includes("Minor Applied Mathematics & Statistics (NEW - 2021 & after)")) {
+            plan.majors[ind] = "Minor Applied Mathematics & Statistics";
+          } 
+        }
+        plan.majors =  [...new Set(plan.majors)];
```
Comment (mpark63) [inline data/replaceOldMajors.js:30]: nice!

File: data/replaceOldMajors.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        // console.log("after: " + plan.majors + "");
+        await plan.save();
+        // console.log("done!\n");
+      }
+      if (plan !== null && plan.major_ids != null && plan.major_ids.length > 0) {
+        console.log("before: " + plan.major_ids);
```
Comment (mpark63) [inline data/replaceOldMajors.js:36]: when is it major_ids vs majors again? 

i think it's major_ids for DIS branch or something

File: data/replaceOldMajors.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        console.log("after: " + plan.major_ids + "");
+        await plan.save();
+        console.log("done!\n");
+      }
+    }
+  }
+  console.log("done")
+}
+
+replaceOldWithNew();
```
Comment (mpark63) [inline data/replaceOldMajors.js:63]: did you run this script yet? :0 

dev / prod?

**Review by mpark63 — CHANGES_REQUESTED** [review]
nice! did you run this script?

**Review by mpark63 — CHANGES_REQUESTED** [review]
ok looks good to me, did you try testing yet? 

if not, please run this script with dev database and test the DEV-142 branches locally. Nothing should break hopefully.

File: data/replaceOldMajors.js (hunk 3 above)
Comment (RosaGao) [reply data/replaceOldMajors.js:35]: i do not know... As per the model typing, Plan should only have majors field. But as least for the mock users some of their plans have major_ids (some have both majors and major_ids).

**Comment by RosaGao** [comment]
I ran the script on dev db and it worked. It just goes to each user and removes parentheses after the major names. 

```
before: B.S. Computer Science (NEW - 2021 & after)
after: B.S. Computer Science
done!
```

**Review by mpark63 — APPROVED** [review]
we ran this on both development and production, it worked. 

feel free to merge for future reference 👍

**Review by mpark63 — APPROVED** [review]
lgtm, let's merge for future reference.

**Review by mpark63 — APPROVED** [review]
lgtm! for this PR we were able to talk about the implementation in person but starting with the next PR, pls include a desription ^^

**Review by mpark63 — APPROVED** [review]
lgtm! idk why it's not letting us merge

**Review by uCredit-Dev — APPROVED** [review]
lgtm

**Pull request merged**